src/frontend/pages/enrich.vue

```diff
@@ -336,29 +391,41 @@ export default {
             responseType: 'json'
           })
           this.state = state?.data
-          // TODO use computed
-          if (Number.isInteger(this.state?.createdAt)) {
-            this.time = Math.round((Date.now() - this.state?.createdAt) / 1000)
-          } else {
-            this.time = 0
-          }
         } catch (err) {
           this.$store.dispatch('snacks/error', this.$t('enrich.errorState'))
           return this.errored()
         }
         await new Promise(resolve => setTimeout(resolve, 1000))
       } while (!state?.data?.done)
 
+      this.stopTimer()
       // done
       this.inProcess = false
       this.id = data.id
     },
 
+    startTimer () {
+      this.timer = setInterval(() => {
+        this.time += 1
+      }, 1000)
+    },
+
+    stopTimer () {
+      clearInterval(this.timer)
+    },
+
+    getColorExtensionFile (extensionSelected) {
+      const extension = this.authorizedFile.find(
+        file => file.name === extensionSelected
```

Comment:
Comparing filename with extension ? I guess some variables must be misnamed here.